engine/runtime/src/main/java/org/enso/interpreter/node/expression/foreign/CoerceNothing.java

```diff
@@ -22,8 +29,22 @@ public static CoerceNothing build() {
   public abstract Object execute(Object value);
 
   @Specialization(guards = "interop.isNull(value)")
-  public Object doNothing(Object value, @CachedLibrary(limit = "1") InteropLibrary interop) {
-    return EnsoContext.get(this).getBuiltins().nothing();
+  public Object doNothing(
+      Object value,
+      @CachedLibrary(limit = "1") InteropLibrary interop,
+      @CachedLibrary(limit = "3") WarningsLibrary warnings) {
+    var nothing = EnsoContext.get(this).getBuiltins().nothing();
+
+    if (nullWarningProfile.profile(warnings.hasWarnings(value))) {
+      try {
+        Warning[] attachedWarnings = warnings.getWarnings(value, null);
+        return new WithWarnings(nothing, attachedWarnings);
```

Comment:
Why do we have the same code twice? Once in `HostValueToEnsoNode` and once in `CoerceNothing`?

Reply:
[`2c3ed37` (#5677)](https://github.com/enso-org/enso/pull/5677/commits/2c3ed37f9573a6862ed7563b308164dea172604a)

Does that look good?